pkg/gardenlet/controller/managedseed/actuator.go

```diff
@@ -561,11 +561,11 @@ func (a *actuator) reconcileSeedSecrets(ctx context.Context, log logr.Logger, sp
 				Namespace: managedSeed.Annotations[seedmanagementv1alpha1constants.AnnotationSeedSecretNamespace],
 			},
 		)
-		if err != nil {
+		if client.IgnoreNotFound(err) != nil {
 			return fmt.Errorf("failed getting secret in the managedseed annotations: %w", err)
 		}
 
-		if metav1.IsControlledBy(secret, managedSeed) {
+		if err == nil && metav1.IsControlledBy(secret, managedSeed) {
```

Comment:
Is it possible to add unit test to cover the case?

Reply:
I added an integration test, PTAL!